src/fft.rg

```diff
@@ -279,48 +588,148 @@ function fft.generate_fft_interface(itype, dtype)
+    end
+  end
+
+  --- Execute plan. This is the __inline version of the task should the user wish to use that.
+  -- @param input Input to execute plan on.
+  -- @param output Output of the executed plan.
+  -- @param plan Plan object used to execute plan.
+  -- @note For GPU, it calls cufftExecZ2Z.
+  -- @note For CPU, it calls fftw_execute_dft.
   __demand(__inline)
-  task iface.execute_plan(input : region(ispace(itype), dtype),
-                          output : region(ispace(itype), dtype),
+  task iface.execute_plan(input : region(ispace(itype), dtype_in),
+                          output : region(ispace(itype), dtype_out),
                           plan : region(ispace(int1d), iface.plan))
-  where reads(input, plan), writes(output) do
+  where reads writes (input, output, plan) do
     var p = iface.get_plan(plan, true)
-    var input_base = get_base(rect_t(input.ispace.bounds), __physical(input)[0], __fields(input)[0])
-    var output_base = get_base(rect_t(output.ispace.bounds), __physical(output)[0], __fields(output)[0])
+    var address_space = c.legion_processor_address_space(get_executing_processor(__runtime()))
+    var input_base = get_base_in(rect_in_t(input.ispace.bounds), __physical(input)[0], __fields(input)[0]).base
+    var output_base = get_base_out(rect_out_t(output.ispace.bounds), __physical(output)[0], __fields(output)[0]).base
 
-    var proc = get_executing_processor(__runtime())
-    if c.legion_processor_kind(proc) == c.TOC_PROC then
-      c.printf("execute plan via cuFFT\n")
-    else
-      c.printf("execute plan via FFTW\n")
-      fftw_c.fftw_execute_dft(p.p, [&fftw_c.fftw_complex](input_base), [&fftw_c.fftw_complex](output_base))
+    fftw_execute_function(p.p, [&fftw_transform_from_type](input_base), [&fftw_transform_to_type](output_base))
+
+    p.address_space = address_space
+
+    rescape
+      if gpu_available then
+        remit rquote
+          if iface.get_num_local_gpus() > 0 then
+            execute_plan_gpu(input, output, plan, p.address_space)
+          end
+        end
+      end
     end
   end
 
-  __demand(__cuda, __leaf)
-  task iface.execute_plan_task(input : region(ispace(itype), dtype),
-                               output : region(ispace(itype), dtype),
+  --- Execute plan - task version. As execute_plan is a __demand(__inline) task, we provide make_plan_task as a wrapper for convenience, should the user wish to use a new task and not an inlined one.
+  -- @param input Input to execute plan on.
+  -- @param output Output of the executed plan.
+  -- @param plan Plan object used to execute plan.
+  task iface.execute_plan_task(input : region(ispace(itype), dtype_in),
+                               output : region(ispace(itype), dtype_out),
                                plan : region(ispace(int1d), iface.plan))
-  where reads(input, plan), writes(output) do
+  where reads writes (input, output, plan) do
```

Comment:
Same note to check the privileges on all of these execute task versions.

Reply:
Changed here: https://github.com/arjunkunna/regent-fft/commit/ba08bf54876149a276a03d839b6e85f4f7e5a7e8